### PR TITLE
Bump bitcoin deps and version

### PR DESCRIPTION
**Pull request by RCasatta**
Required for https://github.com/ElementsProject/elements-miniscript/pull/90

### CODE REVIEW

**Comment by RCasatta** [comment]
I am failing to making `1.58.0` happy and tired trying

I tried various `cargo update` with the version of `Cargo-recent.lock` but I am still failing

**Comment by apoelstra** [comment]
The MSRV of rust-bitcoin is 1.63. You need to bump the MSRV here before you can upgrade rust-bitcoin.

**Comment by uncomputable** [comment]
I bumped the MSRV in #249. It might be worth to merge #249 first and to rebase #247 into #249.

File: simpcli/Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 [package]
 name = "simpcli"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (uncomputable) [inline simpcli/Cargo.toml:3]: 63b2f5ed799a160ccd1c2e83cd2b3dab6c0f3db8: This should be 0.3.0. We forgot to bump the version last time.

**Comment by RCasatta** [comment]
rebased, all green.

I also added a commit adding a `shell.nix` that I used to test rust 1.63, let me know if have to remove it

**Comment by uncomputable** [comment]
We would need to extend the `shell.nix` to make it usable for us. For instance, we require different Rust versions. Let's remove it for now.

**Comment by RCasatta** [comment]
Removed

**Review by apoelstra — APPROVED** [review]
ACK 92e09720deb24b52c80ed397678c9b7b17df0728 successfully ran local tests

**Review by uncomputable — APPROVED** [review]
ACK 92e09720deb24b52c80ed397678c9b7b17df0728

**Pull request merged**